vim_menutrans_helper.py

```diff
@@ -0,0 +1,228 @@
+                            not word_list[index - 1][0].isdigit())):
+                        index += 1
+                    else:
+                        # Menu_item with "ToolBar" has no following words to be 
+                        # translated
+                        if not ("ToolBar." in word_list[index]):
+                            to_be_translated_blocks = word_list[index]
+                        break
+
+                # Remove the redundant digit added at the head and tail of the 
+                # mid_part
+                if (len(to_be_translated_blocks) > 0 and 
+                        to_be_translated_blocks[0] == '5'):
+                    to_be_translated_blocks = to_be_translated_blocks[1:]
+                if (len(to_be_translated_blocks) > 0 and 
+                        to_be_translated_blocks[-1] == '5'):
+                     to_be_translated_blocks = to_be_translated_blocks[:-1]
+
+                # Menu_item with "ToolBar" constructed a tobe_translated_blocks 
+                # with length equals zero
+                if len(to_be_translated_blocks) == 0:
+                    continue
+
+                # Untranslated word string split by "." but with escaped 
+                # literal dot character"\." together
+                to_be_translated_words = []
+                for match in re.finditer(r"(?:\\.|[^.])+", 
+                        to_be_translated_blocks):
+                    word = to_be_translated_blocks[match.start(): match.end()]
+                    if not (len(word) != 0 and word[0] == '-'):
+                        to_be_translated_words.append(word)
+
+                # Conver tobe_translated_words list to string
+                for word in to_be_translated_words:
+                    if len(word) > 0:
+                        untranslated_dict[word.lower()] = (line_number, 
+                                untranslation_file, word)
+            elif word_list[0] in toolbar_commands:
+                untranslated_dict[word_list[1].lower()] = (line_number, 
+                        untranslation_file, word_list[1])
+            elif word_list[0] == "let" and (len(word_list) > 1 and 
+                    word_list[1].startswith("g:menutrans_")):
+                untranslated_dict[word_list[1].lower()] = (line_number, 
+                        untranslation_file, word_list[1])
+
+    # PopUp doesn't need to be translated
+    untranslated_dict.pop("popup", None)
+
+    # String starting and ending with '\ 'or String contains only single digit 
+    # should be skipped.
+    temp_untranslated_dict = {}
+    for u in untranslated_dict.keys():
+        if (u.startswith('\\ ') and u.endswith('\\ ')) or (len(u) == 1 and 
+                u[0] >= '0' and u[0] <= '9'):
+            temp_untranslated_dict[u] = untranslated_dict[u]
+
+    for key in temp_untranslated_dict.keys():
+        if key in untranslated_dict:
+            del untranslated_dict[key]
```

Comment:
Why do you have to create another dictionary to store the keys to be deleted, then do the deletion in another loop? Why can't you directly delete the keys from your original dictionary directly at line 148?